src/services/realtime_monitor.py

```
@@ -220,58 +220,108 @@ def add_data_spec(self, data_spec: str) -> bool:
+                        if ret_code == -1:  # file switch
+                            continue
+                        if ret_code > 0 and buff:
+                            try:
+                                result = updater.process_record(buff)
+                                with self._lock:
+                                    if result:
+                                        self.status.records_imported += 1
+                                        spec_records += 1
+                                    else:
+                                        self.status.records_failed += 1
+                            except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)